File: app/src/main/java/com/code/demo/demo/BannerImageView.java
package com.code.demo.demo;

import android.content.Context;
import android.util.AttributeSet;

import com.code.fastframe.baseview.BaseViewLayout;
import com.code.fastframe.utils.ImageUtils;
import com.code.demo.R;
import com.facebook.drawee.view.SimpleDraweeView;

public class BannerImageView extends BaseViewLayout {


    SimpleDraweeView image;

    public BannerImageView(Context context) {
        super(context);
    }

    public BannerImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public BannerImageView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public int setContentLayout() {
        return R.layout.view_banner_image;
    }

    @Override
    public void initView() {
        image = (SimpleDraweeView) findView(R.id.image);
    }


    @Override
    public void setViewData(Object data) {
        ImageUtils.getInstance().loadImage(image, data + "");
    }
}

File: app/src/main/java/com/code/demo/demo/BannerImageView.kt
package com.code.demo.demo

import android.content.Context
import android.util.AttributeSet
import com.code.demo.R
import com.code.demo.R.layout
import com.code.fastframe.baseview.BaseViewLayout
import com.code.fastframe.utils.ImageUtils
import com.facebook.drawee.view.SimpleDraweeView
import kotlinx.android.synthetic.main.view_banner_image.view.image

class BannerImageView @JvmOverloads constructor(
  context: Context,
  attrs: AttributeSet? = null,
  defStyleAttr: Int = 0
) : BaseViewLayout(context, attrs, defStyleAttr) {

  override fun setContentLayout(): Int {
    return layout.view_banner_image
  }

  override fun initView() {
  }

  override fun setViewData(data: Any?) {
    ImageUtils.getInstance().loadImage(image, data.toString() + "")
  }
}